### PR TITLE
issue 1331 ohCrap Memory Leak

### PR DESCRIPTION
**Pull request by nmccready**
Putting a cap on the amount of checks for isReayd. Throttle isReady checks to 100ms.

### CODE REVIEW

**Comment by nmccready** [comment]
@cthrax @LeonanCarvalho

File: src/coffee/directives/api/utils/is-ready.coffee [hunk 1]
```diff
@@ -1,39 +1,50 @@
 angular.module('uiGmapgoogle-maps.directives.api.utils')
 .service 'uiGmapIsReady', ['$q', '$timeout', ($q, $timeout) ->
-  ctr = 0
-  proms = []
+  _ctr = 0
+  _proms = []
+  _currentCheckNum = 1
+  _maxCtrChecks = 50 #consider maxing this a const so it can be overriden by users
```
Comment (nmccready) [inline src/coffee/directives/api/utils/is-ready.coffee]: whoops typo

File: src/coffee/directives/api/utils/is-ready.coffee [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
+  _maxCtrChecks = 50 #consider making this a angular const so it can be overriden by users
 
-  promises = ->
-    $q.all proms
+  _promises = ->
+    $q.all _proms
+
+
+  _checkIfReady = (deferred, expectedInstances) ->
+    $timeout ->
```
Comment (nmccready) [inline src/coffee/directives/api/utils/is-ready.coffee]: old ohCrap function slowed down and it now has a cap.

**Comment by nmccready** [comment]
@LeonanCarvalho please test and if you want this fixed faster tell me exactly how to produce the problem you were seeing along with getting the chrome output.

**Comment by cthrax** [comment]
:+1:

**Pull request merged**